trino/dbapi.py

```diff
@@ -78,6 +81,39 @@
 logger = trino.logging.get_logger(__name__)
 
 
+class TimeBoundLRUCache:
+    def __init__(self, capacity: int, ttl_seconds: int):
+        self.capacity = capacity
+        self.ttl_seconds = ttl_seconds
+        self.cache = OrderedDict()
+        self.lock = Lock()
+
+    def get(self, key):
+        with self.lock:
+            if key not in self.cache:
+                return None
+            value, timestamp = self.cache[key]
+            if time() - timestamp > self.ttl_seconds:
+                self.cache.pop(key)
+                return None
+            self.cache.move_to_end(key)
+            self.cache[key] = value, time()
+            return value
+
+    def put(self, key, value):
+        with self.lock:
+            self.cache[key] = value, time()
+            self.cache.move_to_end(key)
+            if len(self.cache) > self.capacity:
+                self.cache.popitem(last=False)
+
+    def __repr__(self):
+        return f"LRUCache(capacity: {self.capacity}, ttl: {self.ttl_seconds} seconds, {self.cache})"
+
+
+legacyPreparedStatementsCache = TimeBoundLRUCache(1024, 3600)
```

Comment:
the name is very misleading. It's caching hosts which only support legacy prepared statements.

Reply:
Renamed.